File: app/src/main/java/com/riders/thelab/ui/youtubelike/YoutubeLikeListAdapter.java
package com.riders.thelab.ui.youtubelike;

import android.content.Context;
import android.view.LayoutInflater;
import android.view.ViewGroup;
import android.view.animation.Animation;
import android.view.animation.AnimationUtils;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import com.riders.thelab.R;
import com.riders.thelab.data.local.model.Video;

import java.util.List;


/**
 * Created by michael on 14/04/2016.
 */
public class YoutubeLikeListAdapter extends RecyclerView.Adapter<YoutubeLikeViewHolder> {

    private final Context context;
    private final List<Video> youtubeList;
    private final YoutubeListClickListener listener;

    // Allows to remember the last item shown on screen
    private int lastPosition = -1;


    public YoutubeLikeListAdapter(Context context, List<Video> youtubeList,
                                  YoutubeListClickListener listener) {

        this.context = context;
        this.youtubeList = youtubeList;

        this.listener = listener;
    }


    @Override
    public int getItemCount() {
        if (null != youtubeList) {
            return youtubeList.size();
        }
        return 0;
    }

    @Override
    public long getItemId(int position) {
        return position;
    }


    @NonNull
    @Override
    public YoutubeLikeViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        return new YoutubeLikeViewHolder(
                context,
                LayoutInflater
                        .from(parent.getContext())
                        .inflate(R.layout.row_youtube_like_item, parent, false));
    }

    @Override
    public void onBindViewHolder(@NonNull final YoutubeLikeViewHolder holder, int position) {

        final Video itemYoutubeVideo = youtubeList.get(position);

        holder.setName(itemYoutubeVideo.getName());
        holder.setDescription(itemYoutubeVideo.getDescription());
        holder.setImage(itemYoutubeVideo.getImageThumb());

        holder.itemCardView.setOnClickListener(v ->
                listener.onYoutubeItemClicked(
                        holder.getImageView(),
                        itemYoutubeVideo,
                        holder.getAdapterPosition()));

        // If the bound view wasn't previously displayed on screen, it's animated
        if (position > lastPosition) {
            Animation animation =
                    AnimationUtils.loadAnimation(context, android.R.anim.slide_in_left);
            holder.itemCardView.startAnimation(animation);
            lastPosition = position;
        }
    }
}

File: app/src/main/java/com/riders/thelab/ui/youtubelike/YoutubeLikeListAdapter.kt
package com.riders.thelab.ui.youtubelike

import android.annotation.SuppressLint
import android.content.Context
import android.view.LayoutInflater
import android.view.ViewGroup
import android.view.animation.AnimationUtils
import androidx.recyclerview.widget.RecyclerView
import com.riders.thelab.data.local.model.Video
import com.riders.thelab.databinding.RowYoutubeLikeItemBinding

class YoutubeLikeListAdapter(
    private val context: Context,
    private val youtubeList: List<Video>,
    private val listener: YoutubeListClickListener
) : RecyclerView.Adapter<YoutubeLikeViewHolder>() {


    // Allows to remember the last item shown on screen
    private var lastPosition = -1

    override fun getItemCount(): Int {
        return youtubeList.size
    }

    override fun getItemId(position: Int): Long {
        return position.toLong()
    }


    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): YoutubeLikeViewHolder {
        val binding: RowYoutubeLikeItemBinding =
            RowYoutubeLikeItemBinding.inflate(
                LayoutInflater.from(parent.context),
                parent,
                false
            )

        return YoutubeLikeViewHolder(context, binding, listener)
    }


    override fun onBindViewHolder(holder: YoutubeLikeViewHolder, @SuppressLint("RecyclerView") position: Int) {
        val itemYoutubeVideo = youtubeList[position]

        holder.bind(itemYoutubeVideo)

        /*holder.viewBinding.cardViewItem.setOnClickListener {
            listener.onYoutubeItemClicked(
                holder.getImageView(),
                holder.getNameTextView(),
                holder.getDescriptionView(),
                itemYoutubeVideo,
                holder.absoluteAdapterPosition
            )
        }*/

        // If the bound view wasn't previously displayed on screen, it's animated
        if (position > lastPosition) {
            val animation = AnimationUtils.loadAnimation(context, android.R.anim.slide_in_left)
            holder.viewBinding.cardViewItem.startAnimation(animation)
            lastPosition = position
        }
    }
}